src/main/java/jdraw/figures/Line.java

```diff
@@ -66,7 +75,7 @@ public void setBoundsOnShape(Line2D shape, Point origin, Point corner) {
 
     @Override
     public Figure clone() {
-        return null;
+        return new Line(this);
```

Comment:
You could strengthen the return type from `Figure` to (in this class) `Line` (and similarly in all other `clone` implementations.

Reply:
That is true. My brain is a bit deadlocked on c#. Strenghtening the return types only just was released in the newest version of it. I hope i get a little more used to it soon!

Reply:
I see, this was an extension in C# 9.0. At least one thing where C# could borrow something from Java, ususally it is the other way round.

Reply:
haha, C# 9 also added records. Beautiful little things! i I think java was quicker and added this in v14
It's always a trade off i guess.
C# has many cool language features built in. For me the major thing that i miss in Java are runtime generics and async and await.  But java on the other hand has the bigger eco system and a bigger community.

Reply:
There are some other interesting differences, e.g. that you can implement interface methods separately, i.e. that you can implement two methods with the same signature inherited from different interfaces differently. And they also got the `case` statement right by demanding that a break is necessary (but still providing a fall-through using a `goto`. And obviously LINQ. Records were are in Java since v14, but in v15 they are still a preview feature (i.e. need to be enabled explicitly).